src/OpenLoco/src/Vehicles/VehicleHead.cpp

```diff
@@ -3651,11 +3651,60 @@ namespace OpenLoco::Vehicles
     }
 
     // 0x004AA625
-    void VehicleHead::sub_4AA625()
+    void VehicleHead::landCrashedUpdate()
     {
-        registers regs;
-        regs.esi = X86Pointer(this);
-        call(0x004AA625, regs);
+        VehicleBase* currentVehicle = reinterpret_cast<VehicleBase*>(this);
+        EntityId nextVehicleId = currentVehicle->id;
+
+        while (currentVehicle)
+        {
+            switch (currentVehicle->getSubType())
+            {
+                case VehicleEntityType::head:
+                    reinterpret_cast<VehicleHead*>(currentVehicle)->updateSegmentCrashed();
+                    break;
+                case VehicleEntityType::vehicle_1:
+                    // calls nullsub_21: empty subroutine
+                    break;
+                case VehicleEntityType::vehicle_2:
+                    // calls nullsub_22: empty subroutine
+                    break;
+                case VehicleEntityType::bogie:
+                    reinterpret_cast<VehicleBogie*>(currentVehicle)->updateSegmentCrashed();
+                    break;
+                case VehicleEntityType::body_start:
+                case VehicleEntityType::body_continued:
+                    reinterpret_cast<VehicleBody*>(currentVehicle)->updateSegmentCrashed();
+                    break;
+                case VehicleEntityType::tail:
+                    // calls nullsub_23: empty subroutine
+                    break;
+                default:
+                    break;
+            }
+
+            nextVehicleId = currentVehicle->getNextCar();
+            if (nextVehicleId == EntityId::null)
+            {
+                return;
+            }
+            else
+            {
+                currentVehicle = EntityManager::get<VehicleBase>(nextVehicleId);
+            }
+        }
+    }
+
+    // 0x004AA64B
+    void VehicleHead::updateSegmentCrashed()
+    {
+        _vehicleUpdate_head = X86Pointer(this);
+        _vehicleUpdate_frontBogie = reinterpret_cast<VehicleBogie*>(0xFFFFFFFF);
+        _vehicleUpdate_backBogie = reinterpret_cast<VehicleBogie*>(0xFFFFFFFF);
+
+        VehicleBase* nextCar = EntityManager::get<VehicleBase>(this->getNextCar());
```

Comment:
In general we use `auto train = Vehicle(head);` and then `train.veh1`, `train.veh2` to access the individual parts of the train.